.github/workflows/test-cloud-platforms.yml

```diff
@@ -0,0 +1,149 @@
+        run: |
+          git checkout
+
+          # Prevent releases with -dirty suffix due to forgotten entries in
+          # .gitignore.
+          changes="$(git status --porcelain)"
+          if [ -n "$changes" ] ; then
+            echo "$changes"
+            exit 1
+          fi
+
+          kubectl_gadget=kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}
+
+          CONTAINER_REPO=${{ env.REGISTRY }}/${{ env.CONTAINER_REPO }} \
+          IMAGE_TAG=latest \
+          make $kubectl_gadget
+
+          # Prepare assets for release and actions artifacts
+          platform=$(echo ${kubectl_gadget} | cut -d- -f3-4)
+          mkdir $platform
+          cp $kubectl_gadget $platform/kubectl-gadget
+          cp LICENSE $platform/
+          tar --sort=name --owner=root:0 --group=root:0 \
+            -czf ${kubectl_gadget}.tar.gz -C $platform \
+            kubectl-gadget LICENSE
+          rm -rf $platform
+      - name: Add kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}.tar.gz as artifact.
+        uses: actions/upload-artifact@master
+        with:
+          name: kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}-tar-gz
+          path: /home/runner/work/inspektor-gadget/inspektor-gadget/kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}.tar.gz
+
+  test-integration-aks:
+    name: Integration tests on AKS
+    # level: 1
+    needs: [check-secrets, build-kubectl-gadget]
+    if: needs.check-secrets.outputs.aks == 'true'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os-sku: [Ubuntu]
+        arch: [amd64, arm64]
+    steps:
+    - uses: azure/login@v1
+      name: Login to Azure
+      with:
+        creds: ${{ secrets.AZURE_AKS_CREDS }}
+    - name: Create AKS cluster
+      shell: bash
+      run: |
+        cluster_name=${{ secrets.AZURE_AKS_CLUSTER_PREFIX }}${{ matrix.arch }}-${{ matrix.os-sku }}
+
+        node_size='Standard_D2s_v5'
+        if [ ${{ matrix.arch }} = 'arm64' ]; then
+          node_size='Standard_D2ps_v5'
+        fi
+
+        # Let's keep thing in the US to avoid data crossing the Atlantic.
+        az aks create -l westcentralus -g ${{ secrets.AZURE_AKS_RESOURCE_GROUP }} -n $cluster_name -s $node_size --os-sku ${{ matrix.os-sku }} --no-ssh-key
+    - uses: azure/aks-set-context@v3
+      name: Set AKS cluster context
+      with:
+        cluster-name: ${{ secrets.AZURE_AKS_CLUSTER_PREFIX }}${{ matrix.arch }}-${{ matrix.os-sku }}
+        resource-group: ${{ secrets.AZURE_AKS_RESOURCE_GROUP }}
+        admin: false
+    - uses: actions/checkout@v3
+      with:
+        ref: 'main'
+    - name: Run integration tests
+      uses: ./.github/actions/run-integration-tests
+      env:
+        KUBERNETES_DISTRIBUTION: "aks-${{ matrix.os-sku }}"
+      with:
+        registry: ${{ env.REGISTRY }}
+        container_repo: ${{ env.CONTAINER_REPO }}
+        image_tag: 'latest'
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
As it already succeeded, I suppose we don't need to set any permission for the package for this job.

cc @mqasimsarfraz

Reply:
That is correct. We only needed `package: write` for releases and since we don't trigger this for any tags we should we good.